internal/config/instrumentation.go

```diff
@@ -0,0 +1,412 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package config
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+type (
+	Propagator string
+)
+
+const (
+	// TraceContext represents W3C Trace Context.
+	TraceContext Propagator = "tracecontext"
+	// Baggage represents W3C Baggage.
+	Baggage Propagator = "baggage"
+	// B3 represents B3 Single.
+	B3 Propagator = "b3"
+	// B3Multi represents B3 Multi.
+	B3Multi Propagator = "b3multi"
+	// Jaeger represents Jaeger.
+	Jaeger Propagator = "jaeger"
+	// XRay represents AWS X-Ray.
+	XRay Propagator = "xray"
+	// OTTrace represents OT Trace.
+	OTTrace Propagator = "ottrace"
+	// None represents automatically configured propagator.
+	None Propagator = "none"
+)
+
+// InstrumentationSpec defines the desired state of OpenTelemetry SDK and instrumentation.
+type InstrumentationSpec struct {
```

Comment:
can we not reuse the instrumentation spec from the CRD package?

Reply:
This dependency is mostly incidental though. The CRD itself doesn't depend on the config struct, the webhooks do, and they just happen to be in the same package. There's several things we can do to fix this, like moving webhook implementations to a different package, or having them depend on their own config struct rather than the full operator one.

I would also much rather not define the instrumentation spec twice.